Automattic/wp-calypso
# Editor/CPT: Enable routes for editing custom post types

Closes #3694

This pull request seeks to enable editor routing for custom post types in environments where the `manage/custom-post-types` feature flag is enabled (currently, `development` only).

The following route patterns are now supported:

- `/post/new`
- `/post/example.wordpress.com/new`
- `/post/example.wordpress.com`
- `/post/example.wordpress.com/140`
- `/post/2916284/new`
- `/post/2916284`
- `/post/2916284/140`
- `/page/new`
- `/page/example.wordpress.com/new`
- `/page/example.wordpress.com`
- `/page/example.wordpress.com/140`
- `/page/2916284/new`
- `/page/2916284`
- `/page/2916284/140`
- `/edit/jetpack-portfolio`
- `/edit/example.wordpress.com/jetpack-portfolio`
- `/edit/example.wordpress.com/jetpack-portfolio/140`
- `/edit/2916284/jetpack-portfolio`
- `/edit/2916284/jetpack-portfolio/140`

(For patterns beginning with `/edit`, `jetpack-portfolio` can be substituted with the slug of any post type supported for the site)

__Testing instructions:__

Reference the handy route list above, ensuring that all existing paths continue to behave as expected, with or without the `manage/custom-post-types` flag enabled (Tip: Restart the Calypso process with `DISABLE_FEATURES=manage/custom-post-types make run`)

Verify that the post editor loads custom post types correctly, though do be aware that many labels and functionality (e.g. sidebar elements) are not yet updated to account for custom post types (related: #3702, #3697, #3695).

1. Navigate to [My Sites](http://calypso.localhost:3000/sites)
2. Select a site where a custom post type is available
3. Navigate to a custom post type item in the sidebar
4. Assuming a post exists for the custom post type, click the title of the post
5. Note that the editor loads with the title and content correctly populated
